### PR TITLE
App installation edits

### PR DESCRIPTION
**Pull request by Alec-Shay**
Hi @jhinkey,

Here's a PR with the edits just for the articles pertaining to app installation, since these were a sort of special case we wanted to rework a bit. Essentially, we now have two articles -- one for installing via the file system, and one for installing via the UI -- and it's now left to the introductory article for the section to clue the reader in on which one they should be going to. The file system article in particular also lays out the server restart method first and foremost, and just points out as a note afterward that you can also do so without the restart for auto deployment, as you suggested.

I can only be so sure this meets what everyone was expecting as far as streamlining which goes where, so if you think it still needs to be improved, definitely let me know your thoughts on that. Here are a couple other remaining uncertainties I have with what's here:

- In the file system installation article, as you suggested in the chat with JR (at least if I understood correctly), I added a bit of explanation as to how you should download the app, as well. From how you originally described it, though, it sounded more like you wanted it as a section, or walkthrough for the download as well, which might have seemed out-of-place in an article specifically about installation, so I just made it into a note. Is that sort of like what you were envisioning, or was I way off the mark?

- I named the article for installing via the UI literally "Installing Apps via the UI," though I'm not sure if that is the best. It seemed the most straightforward but maybe there's a better way to word it.

### CODE REVIEW

File: labs/dxp/7.x/installation-and-upgrades/02-setting-up-liferay-dxp/installing-and-managing-apps/installing-apps-via-the-file-system.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Installing Apps via the File System
+
+Installing apps via the file system is the best method for most administrators. Installing via the file system with a system restart is the optimal way to install apps in production-grade environments, as it avoids potential issues that may arise when deploying apps with auto deployment.
```
Comment (jhinkey) [inline labs/dxp/7.x/installation-and-upgrades/02-setting-up-liferay-dxp/installing-and-managing-apps/installing-apps-via-the-file-system.md]: Saying that installing via the file system is best for most administrators doesn't give any reason. One reason we could give is that it's natural to work with the file system once you've downloaded an app as a file. 

Also, saying that it's for "administrators" makes me wonder if this approach is applicable to me (the reader). Do I consider myself an "administrator"? If not, should I skip this approach? Let's let the reader decide whether they like the approach and leave roles out of it. :-D

Here's one way to introduce this topic:

After [downloading an app](TODO) as an LPKG file from the  [Marketplace website], (https://web.liferay.com/marketplace), it's natural to copy it to a local installation. Or if you're working with DXP nodes in a cluster, you can copy the file to each node host and install it. 

Copying the file to your DXP installation while DXP is shutdown is the optimal way to install apps in production-grade environments, as it avoids potential issues that may arise from auto deploying apps. See [Installing Apps via the UI](TODO) for details.

Comment (jhinkey) [reply labs/dxp/7.x/installation-and-upgrades/02-setting-up-liferay-dxp/installing-and-managing-apps/installing-apps-via-the-file-system.md]: @Alec-Shay - I included module JAR and plugin WAR with app LPKG files as they can be deployed the same way as LPKG files.

Comment (jhinkey) [reply labs/dxp/7.x/installation-and-upgrades/02-setting-up-liferay-dxp/installing-and-managing-apps/installing-apps-via-the-file-system.md]: Oh, never mind; I see all three are mentioned in step 2. :-D

Comment (Alec-Shay) [reply labs/dxp/7.x/installation-and-upgrades/02-setting-up-liferay-dxp/installing-and-managing-apps/installing-apps-via-the-file-system.md]: @jhinkey The wording could certainly be improved; I sort of struggled to come up with a good starting sentence (the first bit always seems to be the hardest) and that was the best I came up with, which I agree kind of doesn't really help the reader much.

That said, as far as:

> Let's let the reader decide whether they like the approach and leave roles out of it.

I think part of what we are trying to do is help direct the reader to not leave them with an ambiguous decision of which to use, aren't we? So at least if I'm understanding correctly, we should avoid just presenting decisions as "you could do it this way, if you want, if you like it, (etc.)."... even if, realistically, that is sort of what it comes down to.

File: labs/dxp/7.x/installation-and-upgrades/02-setting-up-liferay-dxp/installing-and-managing-apps/installing-apps-via-the-ui.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Installing Apps via the UI
+
+Installing apps via the UI leverages auto deployment; the app will install (from the LPKG, module JAR, or plugin WAR) while the DXP server is running. Auto deployment copies the app to the appropriate subfolder in `[Liferay Home]/osgi`, converts it to an OSGi bundle (LPKGs and WARs need converting), and deploys the bundle to the OSGi container.
```
Comment (jhinkey) [inline labs/dxp/7.x/installation-and-upgrades/02-setting-up-liferay-dxp/installing-and-managing-apps/installing-apps-via-the-ui.md]: I think it's good to highlight the convenience of installing via the UI up front, and leave the details for the followup sentences.

for example,

You can use DXP's app management UIs to install apps to local and remote DXP servers while they're running. Behind the scenes, DXP copies the app file (Marketplace LPKG, module JAR, or plugin WAR) to the appropriate subfolder in `[Liferay Home]/osgi`, converts it to an OSGi bundle (LPKGs and WARs need converting), and deploys the bundle to the OSGi container. This process is managed by DXP's auto deployment process. 

Apps that aren't designed for auto deployment, however, can cause class loading issues and memory leaks.

**Comment by jhinkey** [comment]
@Alec-Shay - I'm done with the app installation review. Looks really good. 

I think it's worth moving these from labs to docs for JR to review. Thanks for all your work on these!

File: labs/dxp/7.x/installation-and-upgrades/02-setting-up-liferay-dxp/installing-and-managing-apps/installing-apps-via-the-ui.md (hunk 2 above)
Comment (Alec-Shay) [reply labs/dxp/7.x/installation-and-upgrades/02-setting-up-liferay-dxp/installing-and-managing-apps/installing-apps-via-the-ui.md]: @jhinkey I can get behind the general change here. One part I am unsure about though is just this sentence: 

> Apps that aren't designed for auto deployment, however, can cause class loading issues and memory leaks.

I don't think it's good to say this if we don't explain how someone should know whether or not an app is designed for auto deployment. Is there a way we can make this distinction for someone? (I'm not sure myself how one would tell whether or not an app is designed for auto deployment)

Comment (jhinkey) [reply labs/dxp/7.x/installation-and-upgrades/02-setting-up-liferay-dxp/installing-and-managing-apps/installing-apps-via-the-ui.md]: I'll ask the core infra team about determining whether an app is designed for auto-deploy, as I don't know how to determine it.

**Comment by Alec-Shay** [comment]
@jhinkey Thanks for your work on these too, and for reviewing it. I'll shortly push up some more edits to these branches based on your comments.

**Pull request closed without merging**